lib/cfme/cloud_services/data_collector.rb

```diff
@@ -97,11 +100,27 @@ def extract_data(target, model_manifest)
     end
   end
 
+  INSIGHTS_CLIENT_COMMAND = "insights-client".freeze
+  CONTENT_TYPE = "application/vnd.redhat.topological-inventory.something+tgz".freeze
+
   def post_payload(payload)
-    # TODO: Post the payload to the cloud...for now just write to log and STDOUT
-    msg = "Collected the following payload:\n#{JSON.pretty_generate(payload)}"
-    $log.info msg
-    puts msg
+    $log.info "Collected the following payload:\n#{JSON.pretty_generate(payload)}"
+
+    result = nil
+
+    gzipped_tar_file_from(JSON.generate(payload)) do |packed_temporary_file_path|
+      command_params = %W[--payload=#{packed_temporary_file_path} --content-type=#{CONTENT_TYPE}]
```

Comment:
These can be passed to awesome spawn as a hash e.g. `:params => {:payload => packed_temporary_file_path, :content_type => CONTENT_TYPE}` but we can handle that in a followup